script/arrays.js

```diff
@@ -0,0 +1,221 @@
+// minimum
+
+
+// 1. Створи масив «Список покупок». Кожен елемент масиву є об'єктом, який містить назву продукту, 
+// кількість і куплений він чи ні, ціну за одиницю товару, сума. Написати кілька функцій для роботи з таким масивом:
+
+
+function Product(name, quantity, bought, unitPrice) {
+    this.name = name;
+    this.quantity = quantity;
+    this.bought = bought;
+    this.unitPrice = unitPrice;
+    let sum = quantity * unitPrice;
+    this.sum = sum;
+}
+
+let shopingList = new Array();
+shopingList[0] = new Product("apple", 2, false, 5);
+shopingList[1] = new Product("milk", 1, true, 30);
+shopingList[2] = new Product("orange", 5, false, 10);
+shopingList[3] = new Product("sausage", 1, false, 100);
+shopingList[4] = new Product("bread", 1, false, 27);
+shopingList[5] = new Product("lemon", 2, true, 7);
+shopingList[6] = new Product("avocado", 1, false, 55);
+
+//  Виводити весь список на екран таким чином, щоб спочатку йшли продукти, що ще не придбані, а потім - ті, що вже придбали.
+
+shopListOrdered = (arr) => {
+    console.log("Shopping List");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == true) {
+            continue;
+        }
+        console.log(arr[i]);
+    }
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == true) {
+            console.log(arr[i]);
+        }
+    }
+}
+
+shopListOrdered(shopingList);
+
+//  Покупка продукту. Функція приймає назву продукту і відзначає його як придбаний.
+
+productPurchase = (productName) => {
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i].name == productName && shopingList[i].bought == true) {
+            console.log(`You have already bought ${productName}.`);
+            break;
+        }
+        if (shopingList[i].name == productName && shopingList[i].bought == false) {
+            shopingList[i].bought = true;
+            console.log(`You have successfully purchased ${productName}.`);
+            break;
+        } else {
+            console.log(`${productName} is not in the shopping list.`);
+        }
+    }
+}
+
+//  Створення списку (не) придбаних продуктів.
+
+shopListUnpurchased = (arr) => {
+    console.log("Shopping List of Unpurchased Products");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == false) {
+            console.log(arr[i]); 
+            // console.log(arr[i].name); - якщо треба тільки назви продуктів вивести
+        } 
+    }
+}
+
+productPurchase("apple");
+shopListUnpurchased(shopingList);
+
+
+// standart
+
+
+// 1. Видалення продукту зі списку (видалення повинно проводитися шляхом створення нового масиву, 
+// в якому продукт, що ми шукаємо, буде відсутнім)
+
+productRemove = (productName) => {
+    shopingList = shopingList.map((item) => {
+        if (item.name != productName) {
+            return item;
+        }
+    });
+
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i] == undefined) {
```

Comment:
We can simple check `if (!shopingList[i])`. Use implicit type coercion as much as you can inside if statements, we do not need heave calculation inside conditions